java-checks/src/main/java/org/sonar/java/checks/ArchitectureCheck.java

```diff
@@ -84,32 +77,27 @@ private void check(IdentifierTree tree) {
         String fullyQualifiedName = type.fullyQualifiedName();
         Set<String> currentIssues = issues.peekFirst();
         if (!currentIssues.contains(fullyQualifiedName) && WildcardPattern.match(getToPatterns(), fullyQualifiedName)) {
-          reportIssue(tree, shouldCheckId + " must not use " + fullyQualifiedName);
+          context.reportIssue(this, tree, shouldCheckId + " must not use " + fullyQualifiedName);
           currentIssues.add(fullyQualifiedName);
         }
       }
     }
   }
 
-  private void initClass(ClassTree tree) {
+  public void visitClass(ClassTree tree) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1161)